modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -141,6 +152,34 @@ public CompletableFuture<ReplicaResult> processRequest(ReplicaRequest request, S
                 request.groupId(),
                 replicaGrpId);
 
+        if (!waitForActualStateFuture.isDone() && request instanceof PrimaryReplicaRequest) {
+            var targetPrimaryReq = (PrimaryReplicaRequest) request;
+
+            if (request instanceof EmptyPrimaryReplicaRequest) {
+                return waitForActualState(FastTimestamps.coarseCurrentTimeMillis() + 10_000)
+                        .thenComposeAsync(
+                                v -> sendPrimaryReplicaChangeToReplicationGroup(targetPrimaryReq.enlistmentConsistencyToken()),
+                                executor
+                        )
+                        .thenComposeAsync(
+                                unused -> completedFuture(new ReplicaResult(null, null)),
+                                executor
+                        );
+            }
+
+            return placementDriver.addSubgroups(
```

Comment:
As we discussed, this adding must be after the waitForActualState

Reply:
I do not see a way to do this right now.